renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -2372,7 +2372,7 @@ void VulkanReplay::GeneralMisc::Init(WrappedVulkan *driver, VkDescriptorPool des
   VkDescriptorPoolCreateInfo descPoolInfo = {
       VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO,
       NULL,
-      0,
+      VK_DESCRIPTOR_POOL_CREATE_FREE_DESCRIPTOR_SET_BIT,
```

Comment:
I think we should instead have a separate descriptor pool for the pixel history, with a reasonable starting size. That way it will be easier in future to add fallback/expansion if needed (although that seems unlikely), and this pool can still be essentially statically sized since it is only used for a fixed set of descriptor sets.

Reply:
That makes sense. I am currently updating that and I am running into a problem. I am trying to use vkResetDescriptorPool instead of freeing individual sets, but it crashes on vkResetDescriptorPool since GetRecord returns null for the descriptor pool. Do you know if I need to add a record explicitly somewhere? Can't figure out what I am doing wrong

Reply:
Looks like that function has never been called on replay before so it's broken. I think if you wrap all of the first part of the function referring to `record` in `if(IsCaptureMode(m_State))` that should fix it.

Reply:
Makes sense, that worked. Updated now.